projects/ngx-location-picker/src/lib/components/ngx-location-picker.component.ts

```diff
@@ -0,0 +1,391 @@
+import {Component, OnInit, Input, Output, EventEmitter, OnDestroy, HostListener} from '@angular/core';
+import {LeafletMap, baseMapWorldGray, baseMapAntwerp} from '@acpaas-ui/ngx-components/map';
+import {FormBuilder, FormGroup} from '@angular/forms';
+import {NgxLocationPickerService} from '../services/ngx-location-picker.service';
+import {FeatureLayerModel} from '../types/feature-layer.model';
+import {LocationModel} from '../types/location.model';
+import {AddressModel, LatLngModel} from '../types/address.model';
+import {CoordinateModel} from '../types/coordinate.model';
+import {NotificationModel} from '../types/notification.model';
+
+@Component({
+    selector: 'ngx-location-picker',
+    templateUrl: './ngx-location-picker.component.html',
+    styleUrls: ['./ngx-location-picker.component.css']
+})
+export class NgxLocationPickerComponent implements OnInit, OnDestroy {
+
+    /* url to the backend-for-frontend (bff) Should function as pass through to the Location Picker API. */
+    @Input() baseUrl;
+    /* the default zoom level on map load. */
+    @Input() defaultZoom = 14;
+    /* the zoom level when a location is selected. */
+    @Input() onSelectZoom = 16;
+    /* the initial map center on load. */
+    @Input() mapCenter: Array<number> = [51.215, 4.425];
+    /* show a sidebar next to the map leaflet. */
+    @Input() hasSidebar = false;
+    /* show or hide the map. */
+    @Input() showMap = true;
+    /* add layers to show on the map. eg: A-card terminals, Velo stations, ... */
+    @Input() featureLayers: FeatureLayerModel[] = [];
+    /* the input field placeholder text. */
+    @Input() placeholder = 'Locaties zoeken...';
+    /* label to show above the search field. */
+    @Input() label = '';
+    /* label to use when no results were found. */
+    @Input() noResultsLabel = 'Er werden geen locaties gevonden.';
+    /* Optionally set a default location to be shown on the map */
+    @Input() defaultLocation: LatLngModel;
+    /* addPolygon event */
+    @Output() addPolygon = new EventEmitter<any>();
+    /* addLine event */
+    @Output() addLine = new EventEmitter<any>();
+    /* editFeature event */
+    @Output() editFeature = new EventEmitter<any>();
+    /* locationSelect event: fired when selecting a location. */
+    @Output() locationSelect = new EventEmitter<LocationModel | AddressModel | CoordinateModel>();
+
```

Comment:
the old control supported value and valueChange properties. This allows two-way binding the control: `<ngx-location-picker [(value)]="locationValue">

It also implemented the ControlValueAccessor interface so the control could be used in the ngModel style: 

- https://angular.io/api/forms/NgModel
- https://angular.io/guide/forms#two-way-data-binding-with-ngmodel

I think it is an important requirement that you can use it in a form easily as a location field.